backend/graph/api/collection-resolver.go

```diff
@@ -28,16 +30,55 @@ func preloadLoaders(ctx context.Context, loaders *common.BatchLoaders, entries [
 	}
 }
 
-func sectionCollectionEntryResolver(ctx context.Context, loaders *common.BatchLoaders, filteredLoaders *common.FilteredLoaders, section *common.Section, first *int, offset *int) (*utils.PaginationResult[*model.SectionItem], error) {
+func sectionCollectionEntryResolver(
+	ctx context.Context,
+	loaders *common.BatchLoaders,
+	filteredLoaders *common.FilteredLoaders,
+	section *common.Section,
+	first *int,
+	offset *int,
+) (*utils.PaginationResult[*model.SectionItem], error) {
 	if !section.CollectionID.Valid {
 		return &utils.PaginationResult[*model.SectionItem]{}, nil
 	}
 
-	entries, err := collection.GetCollectionEntries(ctx, loaders, filteredLoaders, int(section.CollectionID.ValueOrZero()))
+	collectionId := int(section.CollectionID.ValueOrZero())
+
+	col, err := batchloaders.GetByID(ctx, loaders.CollectionLoader, collectionId)
+	if err != nil {
+		return nil, err
+	}
+
+	entries, err := collection.GetCollectionEntries(ctx, loaders, filteredLoaders, collectionId)
 	if err != nil {
 		return nil, err
 	}
 
+	if col.AdvancedType.Valid {
```

Comment:
Can we turn it around, to something like

```
if !col.AdvancedType.Valid {
     return ....
}
```

?

Reply:
Is `col.AdvancedType.String` of type `string` or `*string`.
If it's `string` then its `""` in case of `!Valid` so you can basically just drop the outside if, as ling as `""` doesn't have a meaning.

Reply:
It doesn't return anything, it just changes the dataset, but i removed the if test as it's not necessary.